### PR TITLE
chore: update nix flake

### PR DESCRIPTION
**Pull request by Alizter**
Looks like treefmt also got an update meaning we have to indent.

### CODE REVIEW

**Comment by ejgallego** [comment]
Thanks @Alizter , indeed let's update the format. How does `treefmt` choose the formatting version tho? It should use package.json right?

**Comment by Alizter** [comment]
Yeah, I guess I'll have to update that too.

**Comment by ejgallego** [comment]
What I mean is that it is strange the formatting has changed, as `treefmt` should read `package.json` and `.ocamlformat` to determine the version of the formatters.

**Comment by Alizter** [comment]
I know that works for ocamlformat but I have no idea how .ts is formatted. I've updated npm and that has seemed to fix it. Should I do something more conservative?

**Comment by ejgallego** [comment]
TS code is formatted using prettier, so indeed, this PR bumps prettier to a new release.

But the weird thing is that the nix Flake should not use the newer prettier version unless we bump package.lock.json

I'm fine updating the prettier version, but IMHO that should usually go on its own PR?

The package.lock update seems a bit too large for my taste, not sure why so much stuff is being installed?

File: editor/code/package-lock.json [hunk 1]
```diff
@@ -424,92 +637,91 @@
...
+      "version": "15.7.12",
+      "resolved": "https://registry.npmjs.org/@types/prop-types/-/prop-types-15.7.12.tgz",
+      "integrity": "sha512-5zvhXYtRNRluoE/jAp4GVsSduVUzNWKkOZrCDBWYtE7biZywwdC2AcEzg+cSMLFRfVgeAFqpfNabiPjxFddV1Q==",
       "dev": true
     },
     "node_modules/@types/react": {
-      "version": "18.2.32",
-      "resolved": "https://registry.npmjs.org/@types/react/-/react-18.2.32.tgz",
-      "integrity": "sha512-F0FVIZQ1x5Gxy/VYJb7XcWvCcHR28Sjwt1dXLspdIatfPq1MVACfnBDwKe6ANLxQ64riIJooXClpUR6oxTiepg==",
+      "version": "18.3.0",
```
Comment (ejgallego) [inline editor/code/package-lock.json:670]: I see a lot of stuff is bumped to newer releases, IMHO we should do that in its own PR.

File: editor/code/package-lock.json [hunk 2]
```diff
@@ -1375,14 +1871,15 @@
       }
     },
     "node_modules/minimatch": {
-      "version": "5.1.6",
-      "resolved": "https://registry.npmjs.org/minimatch/-/minimatch-5.1.6.tgz",
-      "integrity": "sha512-lKwV/1brpG6mBUFHtb7NUmtABCb2WZZmm2wNiOA5hAb8VdCS4B3dtMWyvcoViccwAW/COERjXLt0zP1zXUN26g==",
+      "version": "3.1.2",
```
Comment (ejgallego) [inline editor/code/package-lock.json:1874]: This downgrade is weird

File: editor/code/package-lock.json [hunk 3]
```diff
@@ -1931,9 +2448,9 @@
       }
     },
     "node_modules/typescript": {
-      "version": "5.2.2",
-      "resolved": "https://registry.npmjs.org/typescript/-/typescript-5.2.2.tgz",
-      "integrity": "sha512-mI4WrpHsbCIcwT9cF4FZvr80QUeKvsUsUvKDoR+X/7XHQH98xYD8YHZg7ANtz2GtZt/CBq2QJ0thkGJMHfqc1w==",
+      "version": "5.4.5",
```
Comment (ejgallego) [inline editor/code/package-lock.json:2451]: Another major upgrade here.

**Comment by Alizter** [comment]
@ejgallego Can you update the formatter in it's own PR? I'm not really sure what I'm doing here. I'll rebase the flake update on that afterwards.

File: editor/code/src/client.ts [hunk 4]
```diff
@@ -242,10 +242,10 @@ export function activateCoqLSP(
...
         ? 1
         : callKind == TextEditorSelectionChangeKind.Keyboard
-        ? 2
-        : callKind
-        ? callKind
-        : 3;
+          ? 2
+          : callKind
+            ? callKind
+            : 3;
```
Comment (ejgallego) [inline editor/code/src/client.ts:248]: Looks better yes!

**Comment by Alizter** [comment]
I just wanted to have a more up-to-date development flake, I don't particularly care about the formatting details.

**Comment by ejgallego** [comment]
> @ejgallego Can you update the formatter in it's own PR? I'm not really sure what I'm doing here. I'll rebase the flake update on that afterwards.

Sure, I'm happy to do so; however I'm confused how the Nix stuff determines the version to use in `prettier`.

Can we figure that out first?

Because if `treefmt` doesn't pick the prettier version from package.json we will indeed have problems with a PR updating prettier too.

Indeed, I'm not sure `treefmt` can support non-Nix development use cases.

**Comment by ejgallego** [comment]
> I just wanted to have a more up-to-date development flake, I don't particularly care about the formatting details.

I can see that, but the flake needs to respect the `prettier` version set in package.json.lock , otherwise we will find trouble.

**Comment by ejgallego** [comment]
So it seems the setting should be something like `programs.prettier.package = $(read_version_in_package.lock.json)` IIUC `treefmt` documentation properly.

**Comment by ejgallego** [comment]
Shell code to find the version is:
```
(cd editor/code/ && npm ls --json --package-lock-only | jq '.dependencies.prettier.version')
```

**Comment by ejgallego** [comment]
Another way: `npm pkg get devDependencies.prettier`

Anyways, I think you can just set `programs.prettier.package` to 3.0.3 and forget about it for now.

**Comment by Alizter** [comment]
Supposedly treefmt is going to take care of the version. I'm going to experiment and see why it isn't adhering to it in this case. It could be that I bumped the flake to an unstable version of treefmt-nix or something.

**Comment by ejgallego** [comment]
It seems to me that it is the norm in Nix to have package versions fixed in the hashes, so indeed, it is likely picking the wrong version.

**Comment by ejgallego** [comment]
I think `nix-treefmt` doesn't do any special handling for prettier versions, but I could be wrong, I guess looking at the source code would say.

`treefmt` itself seems to rely on `npm i` doing the right thing.

[So that's our pain point, `npm i` and the flake not being in sync]

**Comment by Alizter** [comment]
https://github.com/prettier/prettier/issues/7940

sigh...

I guess we will have to work out a way to pin this.

**Comment by ejgallego** [comment]
What's wrong with setting `program.prettier.package=` in the flake?

**Comment by Alizter** [comment]
@ejgallego That just let's us set the derivation, but the only one we have is the latest available in nixpkgs where they have tweaked how things are formatted. I can't see a way to set the version in here.
